src/common/tusb_compiler.h

```diff
@@ -44,7 +44,7 @@
 #endif
 
 // Compile-time Assert
-#if __STDC_VERSION__ >= 201112L
+#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
```

Comment:
thanks, cpp has its own version  `static_assert` since C++11, it would be great if you could add ifelf for `__cplusplus` for this as well. It is OK if don't want to do it now, I will do it later on.

https://en.cppreference.com/w/cpp/language/static_assert

```
__cplusplus
This macro is defined when the C++ compiler is in use. You can use __cplusplus to test whether a header is compiled by a C compiler or a C++ compiler. This macro is similar to __STDC_VERSION__, in that it expands to a version number. Depending on the language standard selected, the value of the macro is 199711L for the 1998 C++ standard, 201103L for the 2011 C++ standard, 201402L for the 2014 C++ standard, 201703L for the 2017 C++ standard, or an unspecified value strictly larger than 201703L for the experimental languages enabled by -std=c++2a and -std=gnu++2a.
```

Reply:
Done.